api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java

```diff
@@ -81,55 +82,92 @@ public double getWorkspaceFreeTierBillingUsage(DbWorkspace dbWorkspace) {
 
   /**
    * Check whether users have incurred sufficient cost in their workspaces to trigger alerts due to
-   * passing thresholds or exceeding limits
+   * passing thresholds or exceeding limits. RW-6280 - REQUIRES_NEW transactional mode was added to
+   * make the call to this method create a new transaction with each set of users. In order to
+   * commit the transaction after the call. However, if the user has many workspaces, this method
+   * may still timeout. See {@link #updateFreeTierUsageInBatches(List, List)}
    */
-  public void checkFreeTierBillingUsage() {
-    // retrieve the costs stored in the DB from the last time this was run
-    final Map<DbUser, Double> previousUserCosts = workspaceFreeTierUsageDao.getUserCostMap();
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
+  public void checkFreeTierBillingUsageForUsers(Set<DbUser> users) {
+
+    // Current cost in DB
+    List<WorkspaceCostView> allCostsInDbForUsers = workspaceDao.getWorkspaceCostViews(users);
+    logger.info(String.format("Retrieved %d workspaces from the DB", allCostsInDbForUsers.size()));
+
+    allCostsInDbForUsers =
+        findWorkspaceFreeTierUsagesThatWereNotRecentlyUpdated(allCostsInDbForUsers);
 
-    // retrieve current workspace costs from BigQuery and store in the DB
-    final Map<DbWorkspace, Double> workspaceCosts = getFreeTierWorkspaceCostsFromBQ();
-    workspaceCosts.forEach(workspaceFreeTierUsageDao::updateCost);
+    // No need to proceed since there's nothing to update anyway
+    if (allCostsInDbForUsers.isEmpty()) return;
 
-    // sum current workspace costs by workspace creator
-    final Map<DbUser, Double> userCosts =
-        workspaceCosts.entrySet().stream()
+    List<WorkspaceCostView> workspacesThatRequireUpdate =
+        findActiveOrDeletedWorkspacesThatRequireUpdates(allCostsInDbForUsers);
```

Comment:
emmm, method name is also confusing. is `findActiveOrDeletedWorkspacesThatRequireUpdates` actually just `findWorkspacesThatRequireUpdates`? Do we have other status other than `ActiveOrDeleted`?